### PR TITLE
add sasl support

### PR DESCRIPTION
**Pull request by flyinprogrammer**
This only works when https://github.com/slate/slate-irc/pull/37 is pulled in.
You can edit the package.json if you want ?
Or wait for those merges.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,77 @@
-node_modules/
-npm-debug.log
+### Node template
```
Comment (YaManicKill) [inline .gitignore]: We really don't need all this stuff in the gitignore. Was there something in particular that was causing you issues and wanting to be added to the repo?

File: src/client.js [hunk 2]
```diff
@@ -158,13 +158,31 @@ Client.prototype.connect = function(args) {
...
 
-	if (args.password) {
-		irc.pass(args.password);
-	}
+
 
 	irc.me = nick;
-	irc.nick(nick);
-	irc.user(username, realname);
+	if (!!args.sasl) {
```
Comment (YaManicKill) [inline src/client.js]: Just do if(args.sasl)

Never use "!!" it's very confusing for people who don't know what it means (and there's no real advantage to it. In this case, args.sasl should only ever be a boolean, so theres no need to turn it into a boolean.

Comment (Xe) [inline src/client.js]: With how javascript implicitly typecasts things, it is probably better from a realistic computer science point of view to do this.

Comment (flyinprogrammer) [inline src/client.js]: i am just a script kitty: https://github.com/flyinprogrammer/shout/blob/just_sasl/src/client.js#L191

File: .gitignore (hunk 1 above)
Comment (flyinprogrammer) [inline .gitignore]: i just use intellij, they have a plugin that generates from templates, this is the JetBrains and Node template.

Comment (williamboman) [inline .gitignore]: @flyinprogrammer See https://github.com/slate/slate-irc/pull/37#issuecomment-181505599

**Comment by flyinprogrammer** [comment]
changed to 'args.sasl' and dropped my .gitignore

**Comment by deviousgeek** [comment]
You da man.
`(σ・・)σ :: `

**Comment by MaxLeiter** [comment]
:+1: looks good to me

**Comment by svbeon** [comment]
Pull this in, now

**Comment by flyinprogrammer** [comment]
if you want this code today, it can be used via docker like this:

`docker run -dP flyinprogrammmer/shout:with_flyinprogrammer_slate-irc`

[Dockerfile](https://hub.docker.com/r/flyinprogrammer/shout/~/dockerfile/)

**Comment by svbeon** [comment]
Hmm, it's a bit unclear here where to put the sasl password, maybe change the password box text to "Password/SASL password"

**Pull request closed without merging**